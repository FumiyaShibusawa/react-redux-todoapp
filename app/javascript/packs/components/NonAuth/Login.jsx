import * as React from "react"

class Login extends React.Component {
  render() {
    return(
      <div className="login-form">
        <h1>Login</h1>
        <form action="/sessions" method="POST" charSet="UTF-8">
          <input name="authenticity_token" type="hidden"/>
          <input name="user[name]" type="text" placeholder="Enter your name" autoComplete="username"/>
          <input name="session[email]" type="text" placeholder="Enter your email" autoComplete="username"/>
          <input name="session[password]" type="password" placeholder="Enter your password" autoComplete="current-password"/>
          <button type="submit">Login</button>
        </form>
      </div>
    )
  }
}


export default Login
