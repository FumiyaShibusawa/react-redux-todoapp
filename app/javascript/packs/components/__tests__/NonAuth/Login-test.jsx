import * as React from 'react';
import Login from '../../NonAuth/Login';
import renderer from 'react-test-renderer';

test('it renders correctly', () => {
  const tree = renderer.create(
    <Login />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
