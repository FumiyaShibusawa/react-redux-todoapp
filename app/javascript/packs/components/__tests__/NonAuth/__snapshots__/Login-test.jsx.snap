// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="login-form"
>
  <h1>
    Login
  </h1>
  <form
    action="/sessions"
    charSet="UTF-8"
    method="POST"
  >
    <input
      name="authenticity_token"
      type="hidden"
    />
    <input
      autoComplete="username"
      name="session[name]"
      placeholder="Enter your name"
      type="text"
    />
    <input
      autoComplete="email"
      name="session[email]"
      placeholder="Enter your email"
      type="text"
    />
    <input
      autoComplete="current-password"
      name="session[password]"
      placeholder="Enter your password"
      type="password"
    />
    <button
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
